File: Python/refer-offer/gou_jian_cheng_ji_shu_zu_lcof.py
from typing import *

def constructArr(a: List[int]) -> List[int]:
    b, temp = [1] * len(a), 1
    for i in range(1, len(a)):
        b[i] = b[i - 1] * a[i - 1]
    for i in range(len(a) - 2, -1, -1):
        temp *= a[i + 1]
        b[i] *= temp
    return b

File: core/tmp/Java/src/com/dbc/algorithm/ReferOffer/GouJianChengJiShuZuLcof.java
package com.dbc.algorithm.ReferOffer;

import java.util.Arrays;

public class GouJianChengJiShuZuLcof {
    public int[] constructArr(int[] a) {
        int[] b = new int[a.length];
        int temp = 1;
        Arrays.fill(b, 1);
        for (int i = 1; i < a.length; i++) {
            b[i] = b[i - 1] * a[i - 1];
        }
        for (int i = a.length - 2; i >= 0; i--) {
            temp *= a[i + 1];
            b[i] *= temp;
        }
        return b;
    }
}
